### PR TITLE
fix: prepend https:// in service-account-issuer flag

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

**Reason for Change**:

Fixes test failure in https://testgrid.k8s.io/sig-release-master-informing#aks-engine-azure-windows-master-containerd, specifically the following test case:

```
Kubernetes e2e suite.[sig-auth] ServiceAccounts ServiceAccountIssuerDiscovery should support OIDC discovery of service account issuer [Conformance]
```

Got the following error when executing that test case:
```
2021/02/11 20:32:47 Get "kubernetes.default.svc/.well-known/openid-configuration": unsupported protocol scheme ""
```

I manually [updated some of the API models](https://github.com/kubernetes-sigs/windows-testing/pull/247) used for testing and looks like we need explicitly set the protocol by prepending `https://` in the service-account-issuer flag to fix the test failure.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
are we missing functional aks-engine E2E tests?

**Review by aramase — COMMENTED** [review]
lgtm

What is the plan to switch `http://aka.ms/aks-engine/aks-engine-k8s-e2e.tar.gz` to use the nightly build? If it's not anytime soon, then I'll also need to open a PR to update the dual-stack api models.

**Comment by chewong** [comment]
@aramase @jackfrancis any suggestion on what kind of test we can run here?

**Comment by chewong** [comment]
/hold
doing some additional upstream valdiation

**Comment by chewong** [comment]
The conformance test case above was fixed for containerd + windows cluster but it's still failing for dockershim + windows cluster with the following error:
```
[BeforeEach] [sig-auth] ServiceAccounts
  /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:185
ï¿½[1mSTEPï¿½[0m: Creating a kubernetes client
Mar  2 13:04:08.548: INFO: >>> kubeConfig: /root/tmp206508484/kubeconfig/kubeconfig.eastus2.json
ï¿½[1mSTEPï¿½[0m: Building a namespace api object, basename svcaccounts
ï¿½[1mSTEPï¿½[0m: Binding the e2e-test-privileged-psp PodSecurityPolicy to the default service account in svcaccounts-255
ï¿½[1mSTEPï¿½[0m: Waiting for a default service account to be provisioned in namespace
[It] ServiceAccountIssuerDiscovery should support OIDC discovery of service account issuer [Conformance]
  /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:640
Mar  2 13:04:08.960: INFO: created pod
Mar  2 13:04:08.960: INFO: Waiting up to 5m0s for pod "oidc-discovery-validator" in namespace "svcaccounts-255" to be "Succeeded or Failed"
Mar  2 13:04:08.995: INFO: Pod "oidc-discovery-validator": Phase="Pending", Reason="", readiness=false. Elapsed: 34.811084ms
Mar  2 13:04:11.055: INFO: Pod "oidc-discovery-validator": Phase="Pending", Reason="", readiness=false. Elapsed: 2.094364037s
Mar  2 13:04:13.090: INFO: Pod "oidc-discovery-validator": Phase="Pending", Reason="", readiness=false. Elapsed: 4.129168121s
Mar  2 13:04:15.124: INFO: Pod "oidc-discovery-validator": Phase="Pending", Reason="", readiness=false. Elapsed: 6.163303348s
Mar  2 13:04:17.158: INFO: Pod "oidc-discovery-validator": Phase="Pending", Reason="", readiness=false. Elapsed: 8.197157296s
Mar  2 13:04:19.192: INFO: Pod "oidc-discovery-validator": Phase="Failed", Reason="", readiness=false. Elapsed: 10.231445528s
Mar  2 13:04:49.194: INFO: polling logs
Mar  2 13:04:49.239: INFO: Pod logs: 
2021/03/02 13:04:13 OK: Got token
2021/03/02 13:04:14 OK: got issuer https://kubernetes.default.svc.cluster.local
2021/03/02 13:04:14 Full, not-validated claims: 
openidmetadata.claims{Claims:jwt.Claims{Issuer:"https://kubernetes.default.svc.cluster.local", Subject:"system:serviceaccount:svcaccounts-255:default", Audience:jwt.Audience{"oidc-discovery-test"}, Expiry:1614690849, NotBefore:1614690249, IssuedAt:1614690249, ID:""}, Kubernetes:openidmetadata.kubeClaims{Namespace:"svcaccounts-255", ServiceAccount:openidmetadata.kubeName{Name:"default", UID:"f1f72bc5-b3f2-43ca-95ab-5e42016f49bf"}}}
2021/03/02 13:04:14 Get "https://kubernetes.default.svc.cluster.local/.well-known/openid-configuration": dial tcp: lookup kubernetes.default.svc.cluster.local: no such host
```

still not sure what causes the `dial tcp: lookup kubernetes.default.svc.cluster.local: no such host` error. Tried debugging with a dnsutil container:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: dnsutils
  namespace: default
spec:
  containers:
  - name: dnsutils
    image: k8s.gcr.io/e2e-test-images/jessie-dnsutils:1.4
    command:
      - sleep
      - "3600"
    imagePullPolicy: IfNotPresent
  restartPolicy: Always
```

and then run

```
kubectl exec dnsutils -- nslookup kubernetes.default.svc.cluster.local

Server:		10.0.0.10
Address:	10.0.0.10#53

Name:	kubernetes.default.svc.cluster.local
Address: 10.0.0.1
```

Everything looks good.

**Comment by jsturtevant** [comment]
We identified the issue for Windows and opened a PR to fix the docker scenario in Windows: https://github.com/kubernetes/kubernetes/pull/99860

**Comment by chewong** [comment]
@jackfrancis @aramase added an e2e test case. PTAL.

File: test/e2e/kubernetes/workloads/pod-projected-svc-token.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - image: k8s.gcr.io/e2e-test-images/nginx:1.14-1
```
Comment (chewong) [inline test/e2e/kubernetes/workloads/pod-projected-svc-token.yaml:7]: This is a nginx test image we use upstream, which supports both Linux and Windows.

**Review by aramase — COMMENTED** [review]
/lgtm

**Comment by chewong** [comment]
@jackfrancis can we move forward with this PR. Would love to get this PR merged so we can have green test signal on testgrid. Thanks in advance.

**Pull request merged**